damnit/backend/db.py

```diff
@@ -170,7 +170,10 @@ def add_user_variable(self, variable: UserEditableVariable, exist_ok=False):
 
     def get_user_variables(self):
         user_variables = {}
-        rows = self.conn.execute("SELECT name, title, type, description, attributes FROM variables")
+        rows = self.conn.execute("""
+            SELECT name, title, type, description, attributes FROM variables
+            WHERE type != 'computed'
```

Comment:
For option 2 I'd also prefer if the column was named `gui_type` or something, but it's probably not worth making a breaking change for that.

Reply:
Yeah, if I had been designing the database from scratch, I would probably have made a separate column for this - reusing the `type` column was kind of a bodge to avoid changing the schema again. Adding a column is much simpler than the migration we've just been doing, but still a faff.

So IIUC, your option 2 is logically the same as this, just using `NULL` in place of the string `'computed'`? I'm fine with that, though I don't think it makes much difference to what we can do in the future.

Reply:
Yeah exactly, it's basically just a naming bikeshed but we'd need to add other sentinel values in the future to support injecting variables through the API, and I think it's cleaner to avoid calling them `computed` altogether.

Reply:
OK, I've redone this using `NULL` (aka `None`) as the sentinel.